tests/subsys/dfu/dfu_target_stream/src/main.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <string.h>
+#include <zephyr/types.h>
+#include <stdbool.h>
+#include <ztest.h>
+#include <dfu/dfu_target_stream.h>
+
+#define FLASH_NAME DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL
+#define FLASH_BASE (64*1024)
+#define FLASH_SIZE DT_REG_SIZE(SOC_NV_FLASH_NODE)
+#define FLASH_AVAILABLE (FLASH_SIZE-FLASH_BASE)
+
+#define TEST_ID_1 "test_1"
+#define TEST_ID_2 "test_2"
+
+#define BUF_LEN 14000 /* Note, not page aligned */
+
+static const struct device *fdev;
+static uint8_t sbuf[128];
+static uint8_t read_buf[BUF_LEN];
+static uint8_t write_buf[BUF_LEN] = {[0 ... BUF_LEN - 1] = 0xaa};
+
+#define DFU_TARGET_STREAM_INIT(id_, fdev_, buf_, len_, offset_, size_, cb_)  \
+	dfu_target_stream_init(&(struct dfu_target_stream_init) { .id = id_, \
```

Comment:
Good point. I don't want to have a local variable which is updates, since it makes it harder to separate what has changed between the different calls, I'll see if I can use a static function instead of a macro.